tests/test_libpython.py

```diff
@@ -35,6 +35,6 @@ def test_python_select_by_libpython(
     Tests that profiling of processes running Python, whose basename(readlink("/proc/pid/exe")) isn't "python".
     (for example, uwsgi). We expect to select these because they have "libpython" in their "/proc/pid/maps".
     """
-    profiler = get_python_profiler(1000, 1, Event(), str(tmp_path))
-    process_collapsed = profiler.snapshot()
+    with get_python_profiler(1000, 1, Event(), str(tmp_path), lambda: None) as profiler:
```

Comment:
To avoid `lambda: None` you can make it optional. It handles it as it is right now anyway.

Reply:
k